File: java/using-webhooks/NotificationModel.java
// DocSection: using_webhooks_notification_model
// Tip: Find more about Java SDKs at https://docs.kontent.ai/java
import com.fasterxml.jackson.annotation.JsonProperty;

public class KenticoKontentWebhookModel {
    @JsonProperty("message")
    Message message;

    @JsonProperty("data")
    Data data;

    public Message getMessage() { return message; }

    public Data getData() { return data; }
}

public class Message {
    @JsonProperty("id")
    String id;

    @JsonProperty("type")
    String type;

    @JsonProperty("operation")
    String operation;

    @JsonProperty("api_name")
    String apiName;

    @JsonProperty("project_id")
    String projectId;

    public String getId { return id; }

    public String getType { return type; }

    public String getOperation { return operation; }

    public String getApiName { return apiName; }

    public String getProjectId { return projectId; }
}

public class Data {
    @JsonProperty("items")
    List<Item> items;

    @JsonProperty("taxonomies")
    List<Taxonomy> taxonomies;

    public List<Item> getItems { return items; }

    public List<Taxonomy> getTaxonomies { return taxonomies; }
}

public class Item {
    @JsonProperty("id")
    String id;

    @JsonProperty("codename")
    String codename;

    @JsonProperty("language")
    String language;

    @JsonProperty("type")
    String type;

    public String getId { return id; }

    public String getCodename { return codename; }
    
    public String getLanguage { return language; }

    public String getType { return type; }
}

public class Taxonomy {
    @JsonProperty("id")
    String id;

    @JsonProperty("codename")
    String codename;

    public String getId { return id; }

    public String getCodename { return codename; }
}
// EndDocSection

File: net/using-webhooks/NotificationModel.cs
// DocSection: using_webhooks_notification_model
// Tip: Find more about .NET SDKs at https://docs.kontent.ai/net
using System;
using Newtonsoft.Json;

public class KenticoCloudWebhookModel
{
    [JsonProperty("message")]
    public Message Message { get; set; }

    [JsonProperty("data")]
    public Data Data { get; set; }
}

public class Message
{
    [JsonProperty("id")]
    public Guid Id { get; set; }

    [JsonProperty("type")]
    public string Type { get; set; }

    [JsonProperty("operation")]
    public string Operation { get; set; }

    [JsonProperty("api_name")]
    public string ApiName { get; set; }

    [JsonProperty("project_id")]
    public Guid ProjectId { get; set; }
}

public class Data
{
    [JsonProperty("items")]
    public Item[] Items { get; set; }

    [JsonProperty("taxonomies")]
    public Taxonomy[] Taxonomies { get; set; }
}

public class Item
{
    [JsonProperty("id")]
    public string Id { get; set; }

    [JsonProperty("codename")]
    public string Codename { get; set; }
    
    [JsonProperty("language")]
    public string Language { get; set; }

    [JsonProperty("type")]
    public string Type { get; set; }
}

public class Taxonomy
{
    [JsonProperty("id")]
    public string Id { get; set; }

    [JsonProperty("codename")]
    public string Codename { get; set; }
}
// EndDocSection